### PR TITLE
Added various plug-ins

### PR DESCRIPTION
**Pull request by David-Lam**
* Migrated npm to yarn for better performance
* Added Typescript to be strongly typed
* Added eslint to enforce code style
* Added husky for pre-commit check
* Added commitlint for consistent commit message

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
Comment (aaron-lam) [inline .idea/inspectionProfiles/Project_Default.xml]: Not sure why your `.gitignore` didn't ignore all files under `.idea/`. 
Your team will decide whether we should remove this file or not. Personally I would remove this.

File: .idea/jsLinters/eslint.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/jsLinters/eslint.xml]: Your team will decide whether we should remove this file or not. Personally I would remove this.

File: .idea/misc.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/misc.xml]: Your team will decide whether we should remove this file or not. Personally I would remove this.

File: .idea/modules.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/modules.xml]: Your team will decide whether we should remove this file or not. Personally I would remove this.

File: .idea/workspace.xml [hunk 5]
```diff
@@ -0,0 +1,347 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/workspace.xml]: Remove this.

File: .idea/vcs.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/vcs.xml]: Your team will decide whether remove this or not. Personally I would remove this.

File: .idea/usArmyQuizAppBackend.iml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaron-lam) [inline .idea/usArmyQuizAppBackend.iml]: Remove this.

**Review by aaron-lam — CHANGES_REQUESTED** [review]
The configuration looks good. Nice work!

"Whether we should push `.idea` folder into git repo or not" is a controversial topic. Personally, I don't push `.idea` into git repo. 

Some people would say that sharing `.idea` folder could enforce consistent code style because everyone is using the same Webstorm IDE settings. However, our project uses ESLint to format all code style, so this shouldn't be an issue.

At the very least, `.idea/workspace.xml` and all the `.iml` files should be removed. ([Reference](https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems))

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (David-Lam) [reply .idea/inspectionProfiles/Project_Default.xml]: Removed.

File: .idea/jsLinters/eslint.xml (hunk 2 above)
Comment (David-Lam) [reply .idea/jsLinters/eslint.xml]: Removed.

File: .idea/misc.xml (hunk 3 above)
Comment (David-Lam) [reply .idea/misc.xml]: Removed.

File: .idea/workspace.xml (hunk 5 above)
Comment (David-Lam) [reply .idea/workspace.xml]: Removed.

File: .idea/vcs.xml (hunk 6 above)
Comment (David-Lam) [reply .idea/vcs.xml]: Removed.

File: .idea/usArmyQuizAppBackend.iml (hunk 7 above)
Comment (David-Lam) [reply .idea/usArmyQuizAppBackend.iml]: Removed.

File: .idea/modules.xml (hunk 4 above)
Comment (David-Lam) [reply .idea/modules.xml]: Removed.

**Review by aaron-lam — APPROVED** [review]
The update looks great. Thank you!

**Pull request merged**